### PR TITLE
Bael 7142 spring ai module

### PR DESCRIPTION
**Pull request by Ekatereana**
https://jira.baeldung.com/projects/BAEL/issues/BAEL-7142

### CODE REVIEW

File: spring-ai/postman/spring-ai.postman_collection.json [hunk 1]
```diff
@@ -0,0 +1,56 @@
+{
+	"info": {
+		"_postman_id": "f4282fac-bfe5-45b9-aae6-5ea7c43528ee",
+		"name": "spring-ai",
+		"schema": "https://schema.getpostman.com/json/collection/v2.1.0/collection.json",
+		"_exporter_id": "9576856"
+	},
+	"item": [
+		{
+			"name": "Generate poetry with genre and theme",
```
Comment (vinipx) [inline spring-ai/postman/spring-ai.postman_collection.json:10]: Do we need all these spaces? maybe better identation?

File: spring-ai/src/main/java/com/baeldung/spring/ai/dto/PoetryDto.java [hunk 2]
```diff
@@ -0,0 +1,3 @@
+package com.baeldung.spring.ai.dto;
+
+public record PoetryDto (String title, String poetry, String genre, String theme){}
```
Comment (vinipx) [inline spring-ai/src/main/java/com/baeldung/spring/ai/dto/PoetryDto.java:3]: maybe add a space between **) {}** ?

File: spring-ai/src/main/java/com/baeldung/spring/ai/service/impl/PoetryServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    @Override
+    public String getCatHaiku() {
+        return aiClient.generate(WRITE_ME_HAIKU_ABOUT_CAT);
+    }
+
+    @Override
+    public PoetryDto getPoetryByGenreAndTheme(String genre, String theme) {
+        BeanOutputParser<PoetryDto> poetryDtoBeanOutputParser = new BeanOutputParser<>(PoetryDto.class);
+
+        String promtString = """
```
Comment (vinipx) [inline spring-ai/src/main/java/com/baeldung/spring/ai/service/impl/PoetryServiceImpl.java:35]: typo in prom**p**tString?

File: spring-ai/src/main/resources/application.yml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+spring:
+  ai:
+    openai.api-key: ${OPEN_AI_KEY}
```
Comment (vinipx) [inline spring-ai/src/main/resources/application.yml:3]: minor: aren't we missing an empty line after? for lint verification purposes.. not sure though

File: spring-ai/src/test/resources/application.yml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+spring:
+  ai:
+    openai.api-key: "mock-key-token"
```
Comment (vinipx) [inline spring-ai/src/test/resources/application.yml:3]: minor: aren't we missing an empty line after? for lint verification purposes.. not sure though

File: spring-ai/postman/spring-ai.postman_collection.json [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+					"path": [
+						"ai",
+						"cathaiku"
+					]
+				}
+			},
+			"response": []
+		}
+	]
+}
```
Comment (vinipx) [inline spring-ai/postman/spring-ai.postman_collection.json:56]: minor: aren't we missing an empty line after? for lint verification purposes.. not sure though

File: spring-ai/postman/Spring_AI_Poetry.postman_environment.json [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+		{
+			"key": "theme",
+			"value": "flames",
+			"enabled": true
+		}
+	],
+	"_postman_variable_scope": "environment",
+	"_postman_exported_at": "2023-11-26T19:49:21.755Z",
+	"_postman_exported_using": "Postman/10.20.3"
+}
```
Comment (vinipx) [inline spring-ai/postman/Spring_AI_Poetry.postman_environment.json:25]: minor: aren't we missing an empty line after? for lint verification purposes.. not sure though

File: spring-ai/pom.xml [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+                    <target>16</target>
+                    <excludes>
+                        <exclude>**/*ManualTest.java</exclude>
+                    </excludes>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+
+</project>
```
Comment (vinipx) [inline spring-ai/pom.xml:64]: minor: aren't we missing an empty line after? for lint verification purposes.. not sure though

File: spring-ai/.gitignore [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+/target
+/neoDb*
+/data
+/src/main/webapp/WEB-INF/classes
+*/META-INF/*
+
+# Packaged files #
+*.jar
+*.war
+*.ear
```
Comment (vinipx) [inline spring-ai/.gitignore:12]: minor: aren't we missing an empty line after? for lint verification purposes.. not sure though

**Review by vinipx — COMMENTED** [review]
Thanks for the PR - I've added some comments : )

File: spring-ai/postman/spring-ai.postman_collection.json (hunk 1 above)
Comment (Ekatereana) [reply spring-ai/postman/spring-ai.postman_collection.json:10]: It's generated, so as for me it's should be like that

File: spring-ai/postman/Spring_AI_Poetry.postman_environment.json (hunk 7 above)
Comment (Ekatereana) [reply spring-ai/postman/Spring_AI_Poetry.postman_environment.json:25]: It's generated. Do we really need to update them?

File: spring-ai/postman/spring-ai.postman_collection.json (hunk 1 above)
Comment (vinipx) [reply spring-ai/postman/spring-ai.postman_collection.json:10]: its fine, tks

File: spring-ai/postman/Spring_AI_Poetry.postman_environment.json (hunk 7 above)
Comment (vinipx) [reply spring-ai/postman/Spring_AI_Poetry.postman_environment.json:25]: its fine, tks

**Review by vinipx — APPROVED** [review]

**Pull request merged**